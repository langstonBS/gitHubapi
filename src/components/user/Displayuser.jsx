import React from 'react';
import PropTypes from 'prop-types';
import List from '../list/List';

const Displayuser = ({ name, followerCount, followingCount, html_url, repos }) => (
  <>
    <h2>name: {name}</h2>
    <h2>followerCount: {followerCount}</h2>
    <h2>followingCount: {followingCount}</h2>
    <h2>url :{html_url}</h2>
    <List repos={repos}/>
    

  </>
);

Displayuser.propTypes = {
  name: PropTypes.string.isRequired,
  followerCount: PropTypes.string.isRequired,
  followingCount: PropTypes.string.isRequired,
  html_url : PropTypes.string.isRequired
};

export default Displayuser;
