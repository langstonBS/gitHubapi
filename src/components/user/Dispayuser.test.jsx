import React from 'react';
import { shallow } from 'enzyme';
import Displayuser from './Displayuser';

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(< Displayuser name={'myName'} followerCount={'1'} followingCount={'2'} html_url={'https:/github.com'} repos={[{ name: 'myName', html_url: 'url' }, { name: 'myName', html_url: 'url' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
