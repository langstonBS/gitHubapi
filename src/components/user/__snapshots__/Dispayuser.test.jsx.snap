// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<Fragment>
  <h2>
    name: 
    myName
  </h2>
  <h2>
    followerCount: 
    1
  </h2>
  <h2>
    followingCount: 
    2
  </h2>
  <h2>
    url :
    https:/github.com
  </h2>
  <List
    repos={
      Array [
        Object {
          "html_url": "url",
          "name": "myName",
        },
        Object {
          "html_url": "url",
          "name": "myName",
        },
      ]
    }
  />
</Fragment>
`;
